packages/editor-ui/src/components/executions/workflow/WorkflowExecutionAnnotationSidebar.vue

```diff
@@ -0,0 +1,381 @@
+			isTagsEditEnabled: false,
+			appliedTagIds: [] as string[],
+			tagsSaving: false,
+		};
+	},
+	methods: {
+		async onVoteClick(vote: AnnotationVote) {
+			if (!this.activeExecution) {
+				return;
+			}
+
+			// If user clicked on the same vote, remove it
+			// so that vote buttons act as toggle buttons
+			const voteToSet = vote === this.vote ? null : vote;
+
+			try {
+				await this.executionsStore.annotateExecution(this.activeExecution.id, { vote: voteToSet });
+			} catch (e) {
+				this.showError(e, this.$locale.baseText('executionAnnotationView.vote.error'));
+			}
+		},
+		onTagsEditEnable() {
+			this.appliedTagIds = this.tagIds;
+			this.isTagsEditEnabled = true;
+
+			setTimeout(() => {
+				this.tagsEventBus.emit('focus');
+			}, 0);
+		},
+		async onTagsBlur() {
+			if (!this.activeExecution) {
+				return;
+			}
+
+			const current = (this.tagIds ?? []) as string[];
+			const tags = this.appliedTagIds;
+
+			if (!hasChanged(current, tags)) {
+				this.isTagsEditEnabled = false;
+				return;
+			}
+
+			if (this.tagsSaving) {
+				return;
+			}
+
+			this.tagsSaving = true;
+
+			try {
+				await this.executionsStore.annotateExecution(this.activeExecution.id, { tags });
+			} catch (e) {
+				this.showError(e, this.$locale.baseText('executionAnnotationView.tag.error'));
+			}
+
+			this.tagsSaving = false;
+			this.isTagsEditEnabled = false;
+		},
+		onTagsEditEsc() {
+			this.isTagsEditEnabled = false;
+		},
+	},
+});
+</script>
+
+<template>
+	<div
+		ref="container"
+		:class="['execution-annotation-sidebar', $style.container]"
+		data-test-id="execution-annotation-sidebar"
+	>
+		<div :class="$style.section">
+			<div :class="$style.vote">
+				<div>{{ $locale.baseText('generic.rating') }}</div>
+				<VoteButtons :vote="vote" @vote-click="onVoteClick" />
+			</div>
+			<span class="tags" data-test-id="annotation-tags-container">
+				<AnnotationTagsDropdown
+					v-if="isTagsEditEnabled"
+					v-model="appliedTagIds"
+					ref="dropdown"
+					:create-enabled="true"
+					:event-bus="tagsEventBus"
+					:placeholder="$locale.baseText('executionAnnotationView.chooseOrCreateATag')"
+					class="tags-edit"
+					data-test-id="workflow-tags-dropdown"
+					@blur="onTagsBlur"
+					@esc="onTagsEditEsc"
+				/>
+				<div v-else-if="tagIds.length === 0">
+					<span
+						class="add-tag add-tag-standalone clickable"
+						data-test-id="new-tag-link"
+						@click="onTagsEditEnable"
+					>
+						+ {{ $locale.baseText('executionAnnotationView.addTag') }}
+					</span>
+				</div>
+
+				<span
+					v-else
+					class="tags-container"
+					data-test-id="execution-annotation-tags"
+					@click="onTagsEditEnable"
+				>
+					<span v-for="tag in tags" :key="tag.id" class="clickable">
+						<el-tag :title="tag.name" type="info" size="small" :disable-transitions="true">
+							{{ tag.name }}
+						</el-tag>
+					</span>
+					<span class="add-tag-wrapper">
+						<n8n-button
+							class="add-tag"
+							v-bind="{
```

Comment:
Is there any specific reason why we're doing v-bind and not passing the props directly?

Reply:
Nope, only because I got this snippet from storybook :)